JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesSparql.java

```diff
@@ -110,18 +191,154 @@ protected void loadSparqlConfigs(String filepath) throws IOException {
 		}
 
 	}
-    
+
+	/////////////////////////////////
+//	AVERAGE TIME SERIES UTILITIES
+	////////////////////////////////
+
+	/**
+	 * Returns the custom duration object containing the numerical duration value and temporal unit
+	 * associated with the avergae time series iri.
+	 * Returns null if the average time series does not exist or if the given time series is not of average time series type.
+	 * @param tsIRI Average time series
+	 * @return custom duration with numerical duration and temporal unit
+	 */
+	protected CustomDuration getCustomDuration(String tsIRI){
+
+		CustomDuration duration = null;
+		SelectQuery query = Queries.SELECT();
+		Variable value = SparqlBuilder.var("value");
+		Variable unit = SparqlBuilder.var("unit");
+		Variable dur = SparqlBuilder.var("dur");
+		TriplePattern queryPattern_1 = iri(tsIRI).has(hasAveragingPeriod, dur);
+		TriplePattern queryPattern_2 = dur.has(numericDuration, value).andHas(unitType, unit);
+
+		query.select(value, unit).where(queryPattern_1, queryPattern_2).prefix(prefix_time).prefix(prefix_ontology);
+
+		JSONArray result = kbClient.executeQuery(query.getQueryString());
+		if(!result.isEmpty()){
+			duration = new CustomDuration(Double.valueOf(result.getJSONObject(0).getString("value")), result.getJSONObject(0).getString("unit"));
+		}
+
+		return duration;
+	}
+
+	/**
+	 * Check if an averaging period with given temporalUnit and numericDuration already exists in the knowledge graph.
+	 * Returns null if such an averaging period doesn't exist.
+	 * @param temporalUnit unit type of the averaging period
+	 * @param numericValue numerical duration of the averaging period
+	 * @return Averaging period IRI attached to the given temporalUnit and numericDuration in the knowledge graph
+	 */
+	public String getDurationIRI(String temporalUnit, Double numericValue) {
+
+		String durationIRI = null;
+		String queryString = "periodIRI";
+
+		SelectQuery query = Queries.SELECT();
+		Variable periodIRI = SparqlBuilder.var(queryString);
+		TriplePattern queryPattern = periodIRI.has(numericDuration, numericValue).andHas(unitType, iri(temporalUnit));
+
+		query.select(periodIRI).where(queryPattern).prefix(prefix_time);
+
+		JSONArray result = kbClient.executeQuery(query.getQueryString());
+		if(!result.isEmpty()){
+			durationIRI = result.getJSONObject(0).getString(queryString);
+		}
+
+		return durationIRI;
+	}
+
+	/**
+	 * Creates a hashMap containing the (numericDuration, temporalUnit) mapping to averaging period IRI
+	 * for all the averaging period Iris in the kb
+	 * @return hashMap containing (numericDuration, temporalUnit) mapping to averaging period IRI
+	 */
+	public HashMap createDurationIRIMapping(){
```

Comment:
The recommended return type is Map<CustomDuration, String>, HashMap is more like an implementation class

Reply:
https://rules.sonarsource.com/java/RSPEC-1319